conans/client/tools/net.py

```diff
@@ -13,6 +13,8 @@ def get(url, md5='', sha1='', sha256='', destination=".", filename=""):
     """ high level downloader + unzipper + (optional hash checker) + delete temporary zip
     """
     filename = filename or os.path.basename(url)
+    if any(char in url for char in ["?", "="]) and filename in url:
```

Comment:
Move to the first line and check `if not filename and any(char in url for char in ["?", "="])`?